src/modules/search/SearchModule.ts

```diff
@@ -356,19 +362,54 @@ export default class SearchModule implements ReaderModule {
         });
       });
   }
+
+  createSearchHighlight(selectionInfo: ISelectionInfo, color: string) {
+    try {
+      var createColor: any = color;
+      if (TextHighlighter.isHexColor(createColor)) {
+        createColor = TextHighlighter.hexToRgbChannels(createColor);
+      }
+
+      const uniqueStr = `${selectionInfo.rangeInfo.startContainerElementCssSelector}${selectionInfo.rangeInfo.startContainerChildTextNodeIndex}${selectionInfo.rangeInfo.startOffset}${selectionInfo.rangeInfo.endContainerElementCssSelector}${selectionInfo.rangeInfo.endContainerChildTextNodeIndex}${selectionInfo.rangeInfo.endOffset}`;
+      const sha256Hex = SHA256.hash(uniqueStr);
+      const id = "R2_SEARCH_" + sha256Hex;
+
+      var pointerInteraction = false;
+      const highlight: IHighlight = {
+        color: createColor ? createColor : DEFAULT_BACKGROUND_COLOR,
```

Comment:
More snake case? Probably fine, I was just told it's the devil in JS/TS projects haha